### PR TITLE
increasing group by limit to 3

### PR DESCRIPTION
**Pull request by lcouzens**
## Jira Ticket

[COST-####](https://issues.redhat.com/browse/COST-####)

## Description

This change will ...

## Testing

1. Checkout Branch
2. Restart Koku
3. Hit endpoint or launch shell
    1. You should see ...
4. Do more things...

## Notes

...

### CODE REVIEW

File: koku/masu/processor/__init__.py [hunk 1]
```diff
@@ -143,3 +143,10 @@ def is_ingress_rate_limiting_disabled():  # pragma: no cover
     if res:
         LOG.info(log_json(msg="ingress rate limiting disabled"))
     return res
+
+
+def override_customer_group_by_limit(account):  # pragma: no cover
+    """Flag the customer to override max group_by count."""
+    account = convert_account(account)
+    context = {"schema": account}
+    return UNLEASH_CLIENT.is_enabled("cost-management.backend.override_customer_group_by_limit", context)
```
Comment (samdoran) [inline koku/masu/processor/__init__.py:152]: This function could do the work of determining the limit.
```suggestion
def get_customer_group_by_limit(account: str) -> int:  # pragma: no cover
    """Get the group_by limit for an account."""
    limit = config.GROUP_BY_LIMIT  # this is 2
    account = convert_account(account)
    context = {"schema": account}
    if UNLEASH_CLIENT.is_enabled("cost-management.backend.override_customer_group_by_limit", context):
        limit = config.MAX_GROUP_BY
        
    return limit
```

Comment (maskarb) [inline koku/masu/processor/__init__.py:152]: Given that this is an API feature flag, I think from an organizational standpoint, this func should live...not in Masu. But, idk a better place to put it.

File: koku/masu/config.py [hunk 2]
```diff
@@ -70,6 +70,9 @@ class Config:
     # The decimal precision of our database Numeric columns
     REPORTING_DECIMAL_PRECISION = 9
 
+    # Set max group by options
+    MAX_GROUP_BY = ENVIRONMENT.int("MAX_GROUP_BY_OVERRIDE", default=3)
```
[the remark this answers is not in the document]
Comment (samdoran) [reply koku/masu/config.py:74]: I think that makes more sense.

File: koku/api/report/serializers.py [hunk 3]
```diff
@@ -150,6 +152,8 @@ def __init__(self, *args, **kwargs):
 
         if self._opfields:
             add_operator_specified_fields(self.fields, self._opfields)
+        if self.context.get("request"):
+            self.schema = self.context["request"].user.customer.schema_name
```
[the remark this answers is not in the document]
Comment (lcouzens) [reply koku/api/report/serializers.py:156]: thats fair but it was pretty mad at me locally without that. :shrug:

File: koku/masu/processor/__init__.py (hunk 1 above)
Comment (maskarb) [reply koku/masu/processor/__init__.py:152]: We could just move all these helpers into:
https://github.com/project-koku/koku/blob/main/koku/koku/feature_flags.py

File: koku/api/report/serializers.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (lcouzens) [reply koku/api/report/serializers.py:155]: File "/koku/koku/api/report/serializers.py", line 154, in __init__
koku-koku-worker-1  |     self.schema = self.context["request"].user.customer.schema_name
koku-koku-worker-1  | KeyError: 'request'

[the remark this answers is not in the document]
Comment (maskarb) [reply koku/api/report/serializers.py:156]: That is really weird... I am pretty sure in all of the tags serializers where we do this:
https://github.com/project-koku/koku/blob/main/koku/api/tags/serializers.py#L41-L42
is completely unnecessary. This is where the exception is happening, but if you remove those definitions from all the tags serializers, you don't get this KeyError, and the filtering/excluding still works...

**Review by maskarb — APPROVED** [review]

**Pull request merged**